test/xchain.test.ts

```diff
@@ -1,100 +1,76 @@
 import { ethers } from 'ethers';
-import { amountToBigInt, hashCrossChainUserOp, hashUserOp, sign, toBigInt, verifySignature } from '../src';
+import {
+  amountToBigInt,
+  appendXCallData,
+  hashUserOp,
+  sign,
+  toBigInt,
+  verifySignature,
+} from '../src';
 import { UserOperationBuilder } from 'blndgs-userop';
 import { Account } from '../src';
-import { Asset, Intent } from 'blndgs-model';
+import { Asset, Intent, Stake } from 'blndgs-model';
 import { initTest } from './testUtils';
 import { TIMEOUT } from './constants';
+import { hashCrossChainUserOp, verifyCrossChainSignature } from '../src/crosschain';
 
 describe('computeUserOpHash', () => {
   it('should compute a valid hash for a UserOperation', () => {
-    const mockUserOp = {
-      sender: '0x1234567890123456789012345678901234567890',
-      nonce: '1',
-      initCode: '0x',
-      callData: '0x1234',
-      callGasLimit: '1000000',
-      verificationGasLimit: '1000000',
-      preVerificationGas: '1000000',
-      maxFeePerGas: '1000000000',
-      maxPriorityFeePerGas: '1000000000',
-      paymasterAndData: '0x',
-    };
-
-    const mockBuilder = {
-      getOp: jest.fn().mockReturnValue(mockUserOp),
-    } as unknown as UserOperationBuilder;
-    const expectedHash = '0xfb65960f1001da5618a675bb640a3dcf9c2d6446c1dc2cb29dd9cd17ebdc6756';
-    const hash = hashUserOp(1, mockBuilder);
-    expect(hash).toEqual(expectedHash);
+    const userOp = new UserOperationBuilder()
+      .setSender('0x8Ee0051fDb9Bb3e3Ac94faa30d31895FA9A3ADC5')
+      .setNonce(1)
+      .setInitCode('0x')
+      .setCallData('0x')
+      .setCallGasLimit(799488)
+      .setVerificationGasLimit(628384)
+      .setPreVerificationGas(626688)
+      .setMaxFeePerGas(0)
+      .setMaxPriorityFeePerGas(0)
+      .setPaymasterAndData('0x')
```

Comment:
Need specific gas values per chain and maxFeePerGas, maxPriorityFeePerGas for the destination chain.